packages/http-client/src/ceramic-http-client.ts

```diff
@@ -128,11 +129,14 @@ export default class CeramicClient implements CeramicApi {
     return this._docmap[docIdStr] as unknown as T
   }
 
-  async loadDocument<T extends Doctype>(docId: DocID | string): Promise<T> {
+  async loadDocument<T extends Doctype>(docId: DocID | string, opts?: DocOpts, tip?: CID): Promise<T> {
     docId = typeDocID(docId)
     const docIdStr = docId.toString()
     if (!this._docmap[docIdStr]) {
-      this._docmap[docIdStr] = await Document.load(docId, this._apiUrl, this.context, this._config)
+      // TODO if 'tip' is set then this entry in the cache will be at a specific tip and have polling disabled,
```

Comment:
this is the first problem with this approach.  See my longer comment below for the full story